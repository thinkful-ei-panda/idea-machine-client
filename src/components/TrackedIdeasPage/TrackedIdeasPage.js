import React from 'react'
import Results from '../IdeaResults/IdeaResults'
import config from '../../config'

class TrackedIdeasPage extends React.Component{
  state = {
    results: [],
    error:null,
  }

  componentDidMount(){
    const user_id = 1
    fetch(`${config.API_ENDPOINT}/ideas/followed/${user_id}`)
    .then(res => (!res.ok)
    ?res.json().then(e => Promise.reject(e))
    :res.json())
    .then(results => {
      this.setState({results})
    })
    .catch(error => {
      this.setState({error})
    })
  }

  handleUnfollowClick = (e) => {
    console.log('unfollow')
  }

  render(){
    const {results} = this.state
    return (
      <div>Tracked Ideas Page
        {results.length !== 0 && <Results 
        handleUnfollowClick = {this.handleUnfollowClick}
        results={results} />}
      </div>
    )
  }  
}

export default TrackedIdeasPage