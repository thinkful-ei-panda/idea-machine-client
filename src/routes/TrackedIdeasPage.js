import React from 'react';
import TrackedIdeas from '../components/TrackedIdeas/TrackedIdeas';

class TrackedIdeasPage extends React.Component {

  render() {
    return (
      <div className='TrackedIdeasPage'>
        <h2 hidden>Followed Ideas</h2>
        <TrackedIdeas {...this.props}/>
      </div>
    );
  }
}

export default TrackedIdeasPage
