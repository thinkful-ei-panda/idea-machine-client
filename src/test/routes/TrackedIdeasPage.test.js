import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import TrackedIdeasPage from '../../routes/TrackedIdeasPage';
describe(`TrackedIdeasPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <TrackedIdeasPage />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <TrackedIdeasPage />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})