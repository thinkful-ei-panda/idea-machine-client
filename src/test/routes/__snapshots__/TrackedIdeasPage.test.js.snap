// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackedIdeasPage Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="TrackedIdeasPage"
>
  <h2
    hidden={true}
  >
    Followed Ideas
  </h2>
</div>
`;
